### DISCUSSION TITLE
Incremental Static Regenration without API call Next13

### DISCUSSION THREAD

**Discussion by lsvargas** [vercel/next.js#42692] [Help]
I want to be able to use Incremental static regeneration when calling an endpoint in my Nexjs 13 app. The problem is that my endpoint is an Edge function so I get the following error when building my code:

[![enter image description here][1]][1]

This error is displayed when I try to call my Edge function like this in the `app/` directory.
```javascript
async function fetchPosts() {
  const res = await fetch(`${basePath}/api/myExpensiveEndpoint`, { next: { revalidate: 60 } });
  const data = await res.json();

  return data.posts;
}
``` 

I think this is expected because its a Nextjs Edge function and they are not allow to be called in Static Generation methods. The solution should be calling the method directly instead of the API. Something like this:

```javascript
import { myExpensiveFunction } from '../../lib/functions'

export async function getStaticProps() {
  const data = await myExpensiveFunction()

  return {
    props: { posts: data.posts },
    revalidate: 60,
  };
}
```
But Nexjs13 does not support `getStaticProps` in the new `app/` directory so that would not work. So I was trying to look for a work around this without using the old `pages/` directory.

  [1]: https://i.stack.imgur.com/PMZlR.png

**Comment by lsvargas** [comment]
```javascript
import { fetchExpensiveCalculation } from '../../lib/expensiveCalculation'

export const revalidate = 3600;

async function ExampleComponent () {
  const schedule = await fetchExpensiveCalculation()
}
```

Calling my function and exporting the constant `revalidate` did the trick :)

Here is the source: https://beta.nextjs.org/docs/data-fetching/fetching#segment-cache-configuration

**Reply by icyJoseph** [reply]
Yeah, because API routes do not work during the build process, they only "wake-up" during the server runtime, `next start`. 

So you just needed, as you did, to call your expensive function as you used to do, and export the segment config option to revalidate.

**Comment by icyJoseph** [comment]
Reposting for visibility.

API routes do not work during the build process, these are only available while there is a server runtime, such as `next dev` or `next start`.

Within the context of `appDir` call your function as you used to do, within `getStaticProps` when using pages, and export the segment config option to revalidate.
